File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        for(int i = 0; i < phone_book.length; i++) {
            if(i != phone_book.length - 1) {
              if(phone_book[i + 1].indexOf(phone_book[i]) == 0) {
                  answer = false;
                  break;
              }
            }
        }
        
       
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
   let result = true;
   phone_book = phone_book.sort();
   for(let i = 0; i < phone_book.length; i++) {
       if(i < phone_book.length - 1) {
           if(phone_book[i + 1].indexOf(phone_book[i]) === 0) {
               result = false;
               break;
           }
       }
   }
   return result;
}